File: product_catalog/app/schema.py
from ariadne import gql

type_defs = gql("""
    type Query {
        getProduct(id: ID!): Product
        listProducts: [Product]
    }

    type Mutation {
        createProduct(name: String!, price: Float!): Product
        updateProduct(id: ID!, name: String, price: Float): Product
        deleteProduct(id: ID!): Boolean
    }

    type Product {
        id: ID!
        name: String!
        price: Float!
    }
""")

File: catalog/readProductService/src/schema/schema.js
const { gql } = require("apollo-server");

/**
 * GraphQL type definitions.
 */
const typeDefs = gql`
  type Query {
    listProducts: [Product]
  }

  type Product {
    id: ID!
    name: String!
    description: String
    price: Float!
    stock: Int!
    color: String
    size: String
    material: String
    brand: String
    gender: String
    image_url: String  # Nuevo campo para la URL de la imagen
  }
`;

module.exports = typeDefs;
